autogen/agentchat/groupchat.py

```diff
@@ -178,6 +178,18 @@ def agent_by_name(self, name: str) -> Agent:
         """Returns the agent with a given name."""
         return self.agents[self.agent_names.index(name)]
 
+    def all_agents(self) -> Dict[str, Agent]:
```

Comment:
As far as I see, there is no guarantee that agent names are unique. However, this API assumes they are. In case of collision (two distinct agents having the same name), this function will not return both of them. It would be better to return a list of agents.